pkg/k8s/apis/cilium.io/v2/clrp_types.go

```diff
@@ -0,0 +1,144 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package v2
+
+import (
+	"github.com/cilium/cilium/pkg/policy/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="ciliumlocalredirectpolicy",path="ciliumlocalredirectpolicies",scope="Namespaced",shortName={clrp}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+
+// CiliumLocalRedirectPolicy is a Kubernetes Custom Resource that contains a
+// specification to redirect traffic locally within a node.
+type CiliumLocalRedirectPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Specification of the desired behavior of the local redirect policy
+	Spec CiliumLocalRedirectPolicySpec `json:"spec,omitempty"`
+
+	// Most recent status of the local redirect policy
+	// Read-only
+	//
+	// +deepequal-gen=false
+	// +kubebuilder:validation:Optional
+	Status CiliumLocalRedirectPolicyStatus `json:"status"`
+}
+
+type L3L4Info struct {
+	// ToIp is a destination IP address for traffic to be redirected.
+	//
+	// Example:
+	// Any traffic destined to '169.254.169.254' is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToIp string `json:"toIP,omitempty"`
+
+	// ToPort is destination L4 port number and protocol for traffic to be redirected.
+	// If the protocol field is set to Any or unset, redirection will be
+	// enabled for both TCP and UDP traffic.
+	//
+	// Example:
+	// Any traffic destined to port '53' with UDP protocol is redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToPort api.PortProtocol `json:"toPort,omitempty"`
+}
+
+// RedirectFrom is a frontend configuration that determines traffic that needs to be redirected.
+// The configuration must be specified using an L3L4 tuple or a Kubernetes service.
+type RedirectFrom struct {
+	// ToL3L4Info is a destination tuple {IP, port, protocol} that identifies traffic
+	// to be redirected.
+	//
+	// +kubebuilder:validation:Optional
+	ToIpPort L3L4Info `json:"toIpPort,omitempty"`
```

Comment:
```suggestion
	ToIPPort L3L4Info `json:"toIPPort,omitempty"`
```

I believe the convention is to capitalize abbreviated words.

Reply:
nice, this is from golint check i think. Based on agreement, we can enable the same check with linter.